internal/provider/test_utils.go

```diff
@@ -105,19 +100,20 @@ func (c *MockHTTPClient) Delete(path string) ([]byte, diag.Diagnostics) {
 	return body, diags
 }
 
+// Additional interface methods required by ProviderHTTPClient
 func (c *MockHTTPClient) GetWithStatus(path string) ([]byte, diag.Diagnostics, int) {
 	body, diags := c.Get(path)
-	return body, diags, c.httpCode
+	return body, diags, http.StatusOK
 }
 
 func (c *MockHTTPClient) UpdateWithStatus(path string, data io.Reader) ([]byte, diag.Diagnostics, int) {
 	body, diags := c.Update(path, data)
-	return body, diags, c.httpCode
+	return body, diags, http.StatusOK
 }
 
 func (c *MockHTTPClient) DeleteWithStatus(path string) ([]byte, diag.Diagnostics, int) {
 	body, diags := c.Delete(path)
-	return body, diags, c.httpCode
+	return body, diags, http.StatusNoContent
```

Comment:
It looks like these reverse the changes that were made in #131, is that intended?

In #131, all of these were changed to return `c.httpCode` instead of a hard-code status, and that's what's on `main`:

https://github.com/ansible/terraform-provider-aap/blob/27755c86a4d30be0eaa737af7becdc897c1c4b34/internal/provider/test_utils.go#L108-L120